From 4ade338858b64aca11d9bf907c73fec0f8103fd7 Mon Sep 17 00:00:00 2001
From: Tom Marshall <tdm@cyngn.com>
Date: Fri, 12 Dec 2014 16:35:56 -0800
Subject: [PATCH 2/2] sepolicy: New type sdcard_posix for labeled filesystems

 * Need this because sdcard_external is an alias for vfat

Change-Id: I804ebb0fcf643d603b1a02ee7a54e6d5b6b46294

sepolicy: allow vold to mount ext4 sdcard

* Originally in 9a19f575a4c991bf2d7bc2f8f980909910ee4cce vendor/cm

Change-Id: Id95e48d2380aa7a6727be765e3a52ee49d814bcb

Change-Id: I804ebb0fcf643d603b1a02ee7a54e6d5b6b46294
---
 prebuilts/api/29.0/private/compat/28.0/28.0.ignore.cil | 1 +
 prebuilts/api/29.0/public/domain.te                    | 2 +-
 prebuilts/api/29.0/public/file.te                      | 1 +
 prebuilts/api/29.0/public/vold.te                      | 2 ++
 private/compat/28.0/28.0.ignore.cil                    | 1 +
 public/domain.te                                       | 2 +-
 public/file.te                                         | 1 +
 public/vold.te                                         | 2 ++
 8 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/prebuilts/api/29.0/private/compat/28.0/28.0.ignore.cil b/prebuilts/api/29.0/private/compat/28.0/28.0.ignore.cil
index 1e46f712..8208fa32 100644
--- a/prebuilts/api/29.0/private/compat/28.0/28.0.ignore.cil
+++ b/prebuilts/api/29.0/private/compat/28.0/28.0.ignore.cil
@@ -118,6 +118,7 @@
     runas_app_tmpfs
     runtime_service
     sdcard_block_device
+    sdcard_posix
     sensor_privacy_service
     server_configurable_flags_data_file
     simpleperf_app_runner
diff --git a/prebuilts/api/29.0/public/domain.te b/prebuilts/api/29.0/public/domain.te
index f3487018..cf92d897 100644
--- a/prebuilts/api/29.0/public/domain.te
+++ b/prebuilts/api/29.0/public/domain.te
@@ -492,7 +492,7 @@ neverallow * rootfs:file { create write setattr relabelto append unlink link ren
 
 # Restrict context mounts to specific types marked with
 # the contextmount_type attribute.
-neverallow * {fs_type -contextmount_type}:filesystem relabelto;
+neverallow * {fs_type -contextmount_type -sdcard_posix}:filesystem relabelto;
 
 # Ensure that context mount types are not writable, to ensure that
 # the write to /system restriction above is not bypassed via context=
diff --git a/prebuilts/api/29.0/public/file.te b/prebuilts/api/29.0/public/file.te
index b4c77b1e..0f311664 100644
--- a/prebuilts/api/29.0/public/file.te
+++ b/prebuilts/api/29.0/public/file.te
@@ -123,6 +123,7 @@ type shm, fs_type;
 type mqueue, fs_type;
 type fuse, sdcard_type, fs_type, mlstrustedobject;
 type sdcardfs, sdcard_type, fs_type, mlstrustedobject;
+type sdcard_posix, sdcard_type, fs_type, mlstrustedobject;
 type vfat, sdcard_type, fs_type, mlstrustedobject;
 type exfat, sdcard_type, fs_type, mlstrustedobject;
 type debugfs, fs_type, debugfs_type;
diff --git a/prebuilts/api/29.0/public/vold.te b/prebuilts/api/29.0/public/vold.te
index 49523f43..5479c8cd 100644
--- a/prebuilts/api/29.0/public/vold.te
+++ b/prebuilts/api/29.0/public/vold.te
@@ -68,6 +68,8 @@ allow vold devpts:chr_file rw_file_perms;
 allow vold rootfs:dir mounton;
 allow vold sdcard_type:dir mounton; # TODO: deprecated in M
 allow vold sdcard_type:filesystem { mount remount unmount }; # TODO: deprecated in M
+allow vold sdcard_posix:filesystem { relabelto relabelfrom };
+allow vold labeledfs:filesystem relabelfrom;
 allow vold sdcard_type:dir create_dir_perms; # TODO: deprecated in M
 allow vold sdcard_type:file create_file_perms; # TODO: deprecated in M
 
diff --git a/private/compat/28.0/28.0.ignore.cil b/private/compat/28.0/28.0.ignore.cil
index 1e46f712..8208fa32 100644
--- a/private/compat/28.0/28.0.ignore.cil
+++ b/private/compat/28.0/28.0.ignore.cil
@@ -118,6 +118,7 @@
     runas_app_tmpfs
     runtime_service
     sdcard_block_device
+    sdcard_posix
     sensor_privacy_service
     server_configurable_flags_data_file
     simpleperf_app_runner
diff --git a/public/domain.te b/public/domain.te
index f3487018..cf92d897 100644
--- a/public/domain.te
+++ b/public/domain.te
@@ -492,7 +492,7 @@ neverallow * rootfs:file { create write setattr relabelto append unlink link ren
 
 # Restrict context mounts to specific types marked with
 # the contextmount_type attribute.
-neverallow * {fs_type -contextmount_type}:filesystem relabelto;
+neverallow * {fs_type -contextmount_type -sdcard_posix}:filesystem relabelto;
 
 # Ensure that context mount types are not writable, to ensure that
 # the write to /system restriction above is not bypassed via context=
diff --git a/public/file.te b/public/file.te
index b4c77b1e..0f311664 100644
--- a/public/file.te
+++ b/public/file.te
@@ -123,6 +123,7 @@ type shm, fs_type;
 type mqueue, fs_type;
 type fuse, sdcard_type, fs_type, mlstrustedobject;
 type sdcardfs, sdcard_type, fs_type, mlstrustedobject;
+type sdcard_posix, sdcard_type, fs_type, mlstrustedobject;
 type vfat, sdcard_type, fs_type, mlstrustedobject;
 type exfat, sdcard_type, fs_type, mlstrustedobject;
 type debugfs, fs_type, debugfs_type;
diff --git a/public/vold.te b/public/vold.te
index 49523f43..5479c8cd 100644
--- a/public/vold.te
+++ b/public/vold.te
@@ -68,6 +68,8 @@ allow vold devpts:chr_file rw_file_perms;
 allow vold rootfs:dir mounton;
 allow vold sdcard_type:dir mounton; # TODO: deprecated in M
 allow vold sdcard_type:filesystem { mount remount unmount }; # TODO: deprecated in M
+allow vold sdcard_posix:filesystem { relabelto relabelfrom };
+allow vold labeledfs:filesystem relabelfrom;
 allow vold sdcard_type:dir create_dir_perms; # TODO: deprecated in M
 allow vold sdcard_type:file create_file_perms; # TODO: deprecated in M
 
-- 
2.17.1

